File: web-frontend/src/app/shared/models/UpdateDTO/PostUpdateDTO.ts
import {Topic} from '../Topic';
import {Image} from "../Image";
import {TopicDTO} from "../DTO/TopicDTO";
import {ImageDTO} from "../DTO/ImageDTO";

export interface PostUpdateDTO {
  id: string;
  title: string;
  message: string;
  images: ImageDTO[];
  topics: TopicDTO[];
}

File: api/src/main/java/com/schoolForum/api/ressource/post/DTO/PostUpdateDTO.java
package com.schoolForum.api.ressource.post.DTO;

import com.schoolForum.api.model.*;
import com.schoolForum.api.ressource.image.DTO.ImageDTO;
import com.schoolForum.api.ressource.topic.DTO.TopicDTO;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;
import java.util.UUID;

@Data
@NoArgsConstructor
public class PostUpdateDTO {
    private UUID id;
    private String title;
    private String message;
    private List<TopicDTO> topics;
    private List<ImageDTO> images;
}
